unittest/server/src/ivi_controller_uinttests.cpp

```diff
@@ -0,0 +1,3475 @@
+
+    ASSERT_EQ(layer_set_visibility_fake.call_count, 1);
+}
+
+/** ================================================================================================
+ * @test_id             controller_set_surface_opacity_wrongLayoutSurface
+ * @brief               Test case of controller_set_surface_opacity() where get_surface_from_id() fails, return null pointer
+ * @test_procedure Steps:
+ *                      -# Mocking the wl_resource_get_user_data() does return an object
+ *                      -# Calling the controller_set_surface_opacity()
+ *                      -# Verification point:
+ *                         +# wl_resource_post_event() must be called once time
+ */
+TEST_F(ControllerTests, controller_set_surface_opacity_wrongLayoutSurface)
+{
+    SET_RETURN_SEQ(wl_resource_get_user_data, (void**)mp_iviController, 1);
+
+    controller_set_surface_opacity(nullptr, nullptr, 1, 1);
+
+    ASSERT_EQ(wl_resource_post_event_fake.call_count, 1);
+}
+
+/** ================================================================================================
+ * @test_id             controller_set_surface_opacity_success
+ * @brief               Test case of controller_set_surface_opacity() where get_surface_from_id() success, return an object
+ * @test_procedure Steps:
+ *                      -# Mocking the wl_resource_get_user_data() does return an object
+ *                      -# Mocking the get_surface_from_id() does return an object
+ *                      -# Calling the controller_set_surface_opacity()
+ *                      -# Verification point:
+ *                         +# surface_set_opacity() must be called once time
+ */
+TEST_F(ControllerTests, controller_set_surface_opacity_success)
+{
+    SET_RETURN_SEQ(wl_resource_get_user_data, (void**)mp_iviController, 1);
+    struct ivi_layout_surface *l_layout_surface[1] = {(struct ivi_layout_surface *)0xFFFFFFFF};
+    SET_RETURN_SEQ(get_surface_from_id, l_layout_surface, 1);
+
+    controller_set_surface_opacity(nullptr, nullptr, 1, 1);
+
+    ASSERT_EQ(surface_set_opacity_fake.call_count, 1);
+}
+
+/** ================================================================================================
+ * @test_id             controller_set_layer_opacity_wrongLayoutLayer
+ * @brief               Test case of controller_set_layer_opacity() where get_layer_from_id() fails, return null pointer
+ * @test_procedure Steps:
+ *                      -# Mocking the wl_resource_get_user_data() does return an object
+ *                      -# Calling the controller_set_layer_opacity()
+ *                      -# Verification point:
+ *                         +# wl_resource_post_event() must be called once time
+ */
+TEST_F(ControllerTests, controller_set_layer_opacity_wrongLayoutLayer)
+{
+    SET_RETURN_SEQ(wl_resource_get_user_data, (void**)mp_iviController, 1);
+
+    controller_set_layer_opacity(nullptr, nullptr, 1, 1);
+
+    ASSERT_EQ(wl_resource_post_event_fake.call_count, 1);
```

Comment:
should verify wl_resource_post_event_fake.arg1_history and layer_set_opacity_fake.call_count